### PR TITLE
Admin download files

### LINKED ISSUE #217: Files cannot be downloaded from the admin site
**Describe the bug**
Since moving to the Azure storages backend, the links to download the files in the admin site don't work. The link redirects to the a fairly sensible URL, but `azurite` is the root instead of `localhost` when developing locally. Also no account key or SAS token is provided, so you get an authentication error if you swap out the root. 

**To Reproduce**
Click the link in the admin site of any uploaded file. 

**Expected behavior**
The file should be downloaded.

**Proposed solution**
Maybe we can modify the admin to use `generate_sas_token()` for downloading files.

Aha! Link: https://imperial-college-ict.aha.io/requirements/RSE-132-8

### PR DESCRIPTION
**Pull request by dandavies99**
To be merged after #228 

Buttons and links in the admin site have been updated to use the URL+SAS token in a similar way to the frontend views.

An `AdminMediaWidget` uses the `custom_azure.generate_sas_token()` function directly to generate a working link. This widget is used to download raw data, settings and binary files.

For the list view for `ExperimentDataFiles`, the `get_file_link()` function has been modified to redirect to the download raw data file view.

closes #217

### CODE REVIEW

**Review by CWestICL — APPROVED** [review]
Checked and tested, all seems to be working!

File: battDB/admin.py [hunk 1]
```diff
@@ -155,7 +158,32 @@ class DeviceDataInline(common.admin.TabularInline):
...
+        if "Currently" in output[-1]:
+            output[-1] = "<br>" + output[-1].split("<br>")[-1]
+
+        return mark_safe("".join(output))
+
+
 class DataFileInline(admin.StackedInline):
+    formfield_overrides = {
+        FileField: {"widget": AdminMediaWidget},
+    }
```
Comment (AdrianDAlessandro) [inline battDB/admin.py:186]: I'm guessing this is the trick to override the default widget used in the admin?

**Review by AdrianDAlessandro — APPROVED** [review]
Looks good to me! I can't confirm for sure that the SAS token is being used, but it appears like a token is in the url when I click "view raw" and "download file" in the admin view.

File: battDB/admin.py (hunk 1 above)
Comment (dandavies99) [reply battDB/admin.py:186]: Yes, exactly. In the end it seemed the path of least resistance.

**Pull request merged**